larpandoracontent/LArTwoDReco/TrackInEMShowerAlgorithm.h

```diff
@@ -48,51 +48,41 @@ class TrackInEMShowerAlgorithm : public pandora::Algorithm
         /**
          *  @brief  Returns the upstream cluster address
          */
-        const pandora::Cluster* GetUpstreamCluster() const;
+        const pandora::Cluster *GetUpstreamCluster() const;
 
         /**
          *  @brief  Returns the downstream cluster address
          */
-        const pandora::Cluster* GetDownstreamCluster() const;
+        const pandora::Cluster *GetDownstreamCluster() const;
 
         /**
          *  @brief  Returns the upstream cluster merge point
          */
-        pandora::CartesianVector GetUpstreamMergePoint() const;
+        const pandora::CartesianVector GetUpstreamMergePoint() const;
```

Comment:
Return const reference, for all cases where the instance actually owns the CartesianVector in question. No problem with giving a const reference to the member away, and is more efficient than copying the object.`const pandora::CartesianVector &GetUpstreamMergePoint() const;`